tornado/httputil.py

```diff
@@ -145,13 +145,22 @@ def add(self, name: str, value: str) -> None:
         """Adds a new value for the given key."""
         norm_name = _normalize_header(name)
         self._last_key = norm_name
-        if norm_name in self:
-            self._dict[norm_name] = (
-                native_str(self[norm_name]) + "," + native_str(value)
-            )
-            self._as_list[norm_name].append(value)
+
+        # Strip leading whitespace from the value
+        value = value.lstrip(' \t')  # Remove leading spaces and tabs
+
+        # Handle Content-Length specifically
```

Comment:
Why? In general, to the extent that Content-Length is a special case, we want to raise exceptions whenever there's anything unusual going on and be very careful about anything but the common cases.

Reply:
If we're adhering to RFC guidelines, the current implementation of the `Content-Length` header should indeed raise exceptions for any irregularities. In respect of your concerns, I have completely reverted the `parse_line` method so that it no longer treats the `Content-Length` header as a special case. Instead, in the `add` method, I will not allow the addition of a `Content-Length` header unless it meets the necessary criteria.

Specifically, if there are multiple `Content-Length` headers, I will only allow the replace if their values are the same. Additionally, I am validating that the `Content-Length` value is a positive integer, which is in line with RFC guidelines and can help mitigate potential security vulnerabilities. 

For further context, you can follow this discussion [here](https://github.com/tornadoweb/tornado/pull/3429#issuecomment-2437934724).